### PR TITLE
Decoder improve

### LINKED ISSUE #37: Generate hash table in decoder instead of switch

### PR DESCRIPTION
**Pull request by derzhavin3016**
- [infra] Moved python ver checking into decoder
- [decoder] Implemented hashmap decoding

Closes #37

### CODE REVIEW

File: src/decoder/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 find_package(
-  Python3
+  Python3 3.10
   COMPONENTS Interpreter
```
Comment (Tako-San) [inline src/decoder/CMakeLists.txt]: Don't we need 'Development' component? It was in component list in root CMakeLists.txt

File: src/decoder/decoder.py [hunk 2]
```diff
@@ -83,7 +88,17 @@ def gen_getbits(bit_dict: BitDict, arg: str = "binInst"):
...
+BRANCH_MNEMONICS = (
+    "beq",
+    "bne",
+    "bge",
+    "bgeu",
+    "blt",
+    "bltu",
+    "jal",
+    "jalr",
+    "ecall",
```
Comment (Tako-San) [inline src/decoder/decoder.py:100]: Redundant comma

**Review by Tako-San — COMMENTED** [review]
Could you use our format_target function to format generated files?

File: src/decoder/decoder.py (hunk 2 above)
Comment (derzhavin3016) [reply src/decoder/decoder.py:100]: It has been added by the formatter

**Review by Tako-San — APPROVED** [review]
LGTM

**Pull request merged**